certbot-dovecot/certbot_dovecot/dovecotparser.py

```diff
@@ -0,0 +1,102 @@
+"""Very low-level dovecot config parser based on pyparsing."""
+from pyparsing import (
+    alphanums, restOfLine,
+    Literal, Forward, Group, LineEnd, Optional,
+    OneOrMore, ParserElement, White, Word, ZeroOrMore
+)
+
+class DovecotParser(object):
+    # pylint: disable=expression-not-assigned
+    """A class that parses dovecot configuration with pyparsing."""
+
+    # pyparsing automatically uses whitespaces as delimeters for tokens.
+    # Here, we're setting default whitespace characters to ` ` and `\t`,
+    # excluding `\n` since a new line is used as a delimeter for dovecot
+    # configuration itself
+    ParserElement.setDefaultWhitespaceChars('')
+
+    newline = LineEnd()
+    same_line_space = Optional(Word(' \t').leaveWhitespace())
+    space = Optional(White().leaveWhitespace())
+
+    # Defining comments
+    comment = space + Literal('#') + restOfLine
+
+    # Defining key and value pair
+    key = Word(alphanums) + same_line_space + (Literal('='))
+
+    value_group = OneOrMore(
+        same_line_space
+        + Optional(Literal(',') + same_line_space)
+        + Word(alphanums)
+    )
+
+    key_value_pair = (
+        space + key + value_group + same_line_space
+    )
+
+    # Defining includes
+    # Order matters. It's a first item match.
+    include = (
+        space
+        + (Literal('!include_try') | Literal('!include'))
+        + OneOrMore(same_line_space
+        + Word(alphanums + "_/-?!"))
+        + same_line_space
```

Comment:
`include` statements seem to only accept a single token-- the filepath. ([reference1](https://github.com/dovecot/core/blob/master/src/config/config-parser.c#L961), [reference2](https://wiki.dovecot.org/ConfigFile#Including_config_files)) Although I may have missed a use case-- so I'm not 100% sure.